src/Workspaces/VisualBasic/Portable/Extensions/TypeSyntaxGeneratorVisitor.vb

```diff
@@ -129,10 +129,19 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions
             Dim hasNames = Not names.IsDefault
 
             Return SyntaxFactory.TupleType(SyntaxFactory.SeparatedList(
-                types.Select(Function(t, i) SyntaxFactory.TupleElement(
-                    If(hasNames AndAlso names(i) IsNot Nothing, SyntaxFactory.IdentifierName(names(i)), Nothing),
-                    If(hasNames AndAlso names(i) IsNot Nothing, SyntaxFactory.Token(SyntaxKind.AsKeyword), Nothing),
-                    t.GenerateTypeSyntax()))))
+                types.Select(Function(t, i)
+                                 Dim element As TupleElementSyntax
```

Comment:
NamedTupleElement and TypedTupleElement do not inherit from each other and inference would not pick common base since it is not something that is in the candidate set

The alternative is to cast one return or another. I think it is uglier.

---
In reply to: [82297349](https://github.com/dotnet/roslyn/pull/14284#discussion_r82297349) [](ancestors = 82297349,82296019)